<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>format_image_to_sce — format_image_to_sce • SPIAT</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="format_image_to_sce — format_image_to_sce" />
<meta property="og:description" content="Formats an INFORM, HALO, Visum, CODEX, cellprofiler outputs into a singlecellexperiment class. Users can also input a generic format.
The count assay stores the intensity level of every marker (rows) for
every cell (columns). Cell phenotype, x and y coordinates and other properties are stored under colData. If INFORM format, the cell properties 
are Cell.Area, Nucleus.Area, Nucleus.Compactness, Nucleus.Axis.Ratio, and Cell.Axis.Ratio. If HALO format, the cell properties are Cell.Area, 
Nucleus.Area and Cytoplasm.Area." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPIAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/introduction.html">Overview of the SPIAT package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cancer-evolution/SPIAT/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>format_image_to_sce</h1>
    <small class="dont-index">Source: <a href='https://github.com/cancer-evolution/SPIAT/blob/master/R/format_image_to_sce.R'><code>R/format_image_to_sce.R</code></a></small>
    <div class="hidden name"><code>format_image_to_sce.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Formats an INFORM, HALO, Visum, CODEX, cellprofiler outputs into a singlecellexperiment class. Users can also input a generic format.
The count assay stores the intensity level of every marker (rows) for
every cell (columns). Cell phenotype, x and y coordinates and other properties are stored under colData. If INFORM format, the cell properties 
are Cell.Area, Nucleus.Area, Nucleus.Compactness, Nucleus.Axis.Ratio, and Cell.Axis.Ratio. If HALO format, the cell properties are Cell.Area, 
Nucleus.Area and Cytoplasm.Area.</p>
    </div>

    <pre class="usage"><span class='fu'>format_image_to_sce</span><span class='op'>(</span>
  format <span class='op'>=</span> <span class='st'>"INFORM"</span>,
  path <span class='op'>=</span> <span class='cn'>NULL</span>,
  markers <span class='op'>=</span> <span class='cn'>NULL</span>,
  locations <span class='op'>=</span> <span class='cn'>NULL</span>,
  dye_columns_interest <span class='op'>=</span> <span class='cn'>NULL</span>,
  intensity_columns_interest <span class='op'>=</span> <span class='cn'>NULL</span>,
  path_to_visium_coordinates <span class='op'>=</span> <span class='cn'>NULL</span>,
  path_to_codex_cell_phenotypes <span class='op'>=</span> <span class='cn'>NULL</span>,
  intensity_matrix <span class='op'>=</span> <span class='cn'>NULL</span>,
  phenotypes <span class='op'>=</span> <span class='cn'>NULL</span>,
  coord_x <span class='op'>=</span> <span class='cn'>NULL</span>,
  coord_y <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>format</th>
      <td><p>String defining the software used for cell segmentation. Options: "INFORM", "HALO", "Visium", "CODEX" "cellprolifer, or "general"</p></td>
    </tr>
    <tr>
      <th>path</th>
      <td><p>String of the path location of either HALO csv file, INFORM text file or CODEX csv file. If 10X Visium transcriptomics data,
path to the "filtered_feature_bc_matrix" folder generated by CellRanger</p></td>
    </tr>
    <tr>
      <th>markers</th>
      <td><p>For INFORM and HALO formats. Vector containing the markers used for staining. Must match the order of the 'markers' parameter</p></td>
    </tr>
    <tr>
      <th>locations</th>
      <td><p>Vector containing the locations of markers used for staining. Location can be either "Nucleus", "Cytoplasm" or "Membrane". 
This is used to select the Intensity column and can be used instead of intensity_columns_interest.</p></td>
    </tr>
    <tr>
      <th>dye_columns_interest</th>
      <td><p>For HALO formats. Use if locations is not specified. Vector of names of the columns with the marker status 
(i.e. those indicating 1 or 0 for whether the cell is positive or negative for the marker). 
Column names must match the order of the 'markers' parameter.</p></td>
    </tr>
    <tr>
      <th>intensity_columns_interest</th>
      <td><p>For HALO formats. Use if locations is not specified. Vector with the names of the columns with the level of each marker.
Column names must match the order of the 'markers' parameter</p></td>
    </tr>
    <tr>
      <th>path_to_visium_coordinates</th>
      <td><p>For 10X Visium data. String of the path to the "tissue_positions_list.csv" generated by CellRanger</p></td>
    </tr>
    <tr>
      <th>path_to_codex_cell_phenotypes</th>
      <td><p>For CODEX data. String of the path to the Cluster ID/Cell type file.</p></td>
    </tr>
    <tr>
      <th>intensity_matrix</th>
      <td><p>For the "general" format. A matrix of marker intensities or gene expression where the column names are the Cell IDs, and the rownames the marker or genes.</p></td>
    </tr>
    <tr>
      <th>phenotypes</th>
      <td><p>A vector of cell phenotypes in the same order in which they appear in intensity_matrix. If no phenotypes available, then a vector of NA can be used as input. Note
that the combination of markers (e.g. CD3,CD4) needs to be used instead of the cell type name (e.g. helper T cells)</p></td>
    </tr>
    <tr>
      <th>coord_x</th>
      <td><p>A vector with the X coordinates of the cells. The cells must be in the same order as in the intensity_matrix.</p></td>
    </tr>
    <tr>
      <th>coord_y</th>
      <td><p>A vector with the Y coordinates of the cells. The cells must be in the same order as in the intensity_matrix.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A SingleCellExperiment object is returned</p>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Anna Trigos, Tom  Yang, Yuzhou  Feng, Volkan  Ozcoban, Maria  Doyle.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


